xdcr/xdcr_test.go

```diff
@@ -50,74 +52,299 @@ func TestMobileXDCRNoSyncDataCopied(t *testing.T) {
+// getTwoBucketDataStores creates two data stores in separate buckets to run xdcr within. Returns a named collection or a default collection based on the global test configuration.
+func getTwoBucketDataStores(t *testing.T) (base.Bucket, sgbucket.DataStore, base.Bucket, sgbucket.DataStore) {
+	ctx := base.TestCtx(t)
+	base.RequireNumTestBuckets(t, 2)
+	fromBucket := base.GetTestBucket(t)
+	t.Cleanup(func() {
+		fromBucket.Close(ctx)
+	})
+	toBucket := base.GetTestBucket(t)
+	t.Cleanup(func() {
+		toBucket.Close(ctx)
+	})
+	var fromDs base.DataStore
+	var toDs base.DataStore
+	if base.TestsUseNamedCollections() {
+		var err error
+		fromDs, err = fromBucket.GetNamedDataStore(0)
+		require.NoError(t, err)
+		toDs, err = toBucket.GetNamedDataStore(0)
+		require.NoError(t, err)
+	} else {
+		fromDs = fromBucket.DefaultDataStore()
+		toDs = toBucket.DefaultDataStore()
+	}
+	return fromBucket, fromDs, toBucket, toDs
+}
 
-		}, time.Second*5, time.Millisecond*100)
-		totalDocsWritten += 2
-		totalDocsFiltered++
-		if base.UnitTestUrlIsWalrus() {
-			// TODO: CBG-3861 implement _vv support in rosmar
-			continue
-		}
-		// in mobile xdcr mode a version vector will be written
-		if base.TestSupportsMobileXDCR() {
-			// verify VV is written to docs that are replicated
-			for _, doc := range []string{normalDoc, attachmentDoc} {
-				require.EventuallyWithT(t, func(c *assert.CollectT) {
-					xattrs, _, err := toDs.GetXattrs(ctx, doc, []string{"_vv"})
-					assert.NoError(c, err, "Could not get doc %s", doc)
-					vvXattrBytes, ok := xattrs["_vv"]
-					require.True(t, ok)
-					var vvXattrVal map[string]any
-					require.NoError(t, base.JSONUnmarshal(vvXattrBytes, &vvXattrVal))
-					assert.NotNil(c, vvXattrVal[version])
-					assert.NotNil(c, vvXattrVal[source])
-					assert.NotNil(c, vvXattrVal[curCAS])
-
-				}, time.Second*5, time.Millisecond*100)
-			}
-		}
+func TestReplicateVV(t *testing.T) {
+	fromBucket, fromDs, toBucket, toDs := getTwoBucketDataStores(t)
+	ctx := base.TestCtx(t)
+	fromBucketSourceID, err := getSourceID(ctx, fromBucket)
+	require.NoError(t, err)
+
+	testCases := []struct {
+		name        string
+		docID       string
+		body        string
+		preXDCRFunc func(t *testing.T, docID string) uint64
+	}{
+		{
+			name:  "normal doc",
+			docID: "doc1",
+			body:  `{"key":"value"}`,
+			preXDCRFunc: func(t *testing.T, docID string) uint64 {
+				cas, err := fromDs.WriteCas(docID, 0, 0, []byte(`{"key":"value"}`), 0)
+				require.NoError(t, err)
+				return cas
+			},
+		},
+		{
+			name:  "dest doc older, expect overwrite",
+			docID: "doc2",
+			body:  `{"datastore":"fromDs"}`,
+			preXDCRFunc: func(t *testing.T, docID string) uint64 {
+				_, err := toDs.WriteCas(docID, 0, 0, []byte(`{"datastore":"toDs"}`), 0)
+				require.NoError(t, err)
+				cas, err := fromDs.WriteCas(docID, 0, 0, []byte(`{"datastore":"fromDs"}`), 0)
+				require.NoError(t, err)
+				return cas
+			},
+		},
+		{
+			name:  "dest doc newer, expect keep old version",
```

Comment:
This says dest doc newer, but looks like fromDs gets written to last?

Reply:
This test was wrong, thanks.